extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/CurrentVertxRequest.java

```diff
@@ -17,6 +21,16 @@ public RoutingContext getCurrent() {
         return current;
     }
 
+    @Produces
+    @RequestScoped
+    public Session getCurrentSession() {
+        // the session might be `null`, so naively, this producer would be `@Dependent`,
+        // but that would break when injected into `@Singleton` or `@ApplicationScoped` beans;
+        // instead, when there's no session, we return a dummy implementation that always throws
+        Session result = current.session();
+        return result != null ? result : DummySession.INSTANCE;
```

Comment:
Hm, what's the difference compared to throwing an exception directly in the producer method? I mean there's no way to use the dummy session at all. And the producer method is called when the injected proxy is actually used. So if I have `@Inject Session session` then `session.id()` would fail in both cases :shrug:.

Reply:
That is probably a good point. I need to think more about it. I admit I wrote this code on Friday afternoon, so it isn't my best for sure :-)

Reply:
Fixed.